framework/master-key-provider-keyring.md

```diff
@@ -0,0 +1,125 @@
+[//]: # (Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.)
+[//]: # (SPDX-License-Identifier: CC-BY-SA-4.0)
+
+# Master Key Provider Keyring
+
+## Version
+
+0.1.0-preview
+
+## Implementations
+
+- [Java](https://github.com/aws/aws-encryption-sdk-java/blob/master/src/main/java/com/amazonaws/encryptionsdk/keyrings/MasterKeyProviderKeyring.java)
+- [Python](https://github.com/aws/aws-encryption-sdk-python/src/aws_encryption_sdk/keyrings/master_key.py)
+
+## Overview
+
+A keyring which wraps a legacy [Master Key Provider](master-key-provider-interface.md) to facilitate transitioning to [Keyrings](./keyring-interface.md).
+
+### Legacy
+
+Master Key Providers are a legacy construct that have been supplanted by [Keyrings](./keyring-interface.md). 
+The Master Key Provider Keyring should only be used when interfacing with Master Key Providers that cannot otherwise be
+transitioned to keyrings.
+
+## Definitions
+
+### Conventions used in this document
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL"
+in this document are to be interpreted as described in [RFC 2119](https://tools.ietf.org/html/rfc2119).
+
+## Inputs
+
+On keyring initialization, a keyring MUST define the following:
+
+- [Master Key Provider](#master-key-provider)
+
+### Master Key Provider
+
+A [master key provider](master-key-provider-interface.md) that can decrypt data keys using available [master keys](master-key-interface.md)
+ as well as provide [master keys](master-key-interface.md) for use in encryption.
```

Comment:
I would leave the definition of a MKP to the MKP spec. Notably, this definition implies that the MKP provided MUST work on both encrypt and decrypt paths, which might not always be correct.

Reply:
I've removed the definition